storage-local/src/main/java/io/kestra/storage/local/LocalStorage.java

```diff
@@ -60,6 +51,24 @@ public boolean exists(String tenantId, URI uri) {
         return Files.exists(getPath(tenantId, uri));
     }
 
+    @Override
+    public List<FileAttributes> list(String tenantId, URI uri) throws IOException {
+        try (Stream<Path> stream = Files.list(getPath(tenantId, URI.create(uri.getPath())))) {
+            return stream
+                .map(throwFunction(file -> {
+                    URI relative = URI.create(
+                        getPath(tenantId, URI.create("")).relativize(
+                            Path.of(file.toUri())
+                        ).toString()
+                    );
+                    return getAttributes(tenantId, relative);
```

Comment:
I think we miss a check here as if `relativize` fail because the child path is not inside the base path, the child path is returned leading to potential path traversal issue.

We should check that `relative`is a child of the basePath and throw and exception if not (ideally to be tested)

Reply:
the path come from the list result I'm not sure this is possible nor how I could test that ?
Also it seems that the relativize method already throws
> Exception: This method throws IllegalArgumentException if other is not a Path that can be relativized against this path.